### DISCUSSION TITLE
How to access nvimtree state? Want to toggle on focus, but not if already open.

### DISCUSSION THREAD

**Discussion by rtviii** [nvim-tree/nvim-tree.lua#1683] [Q&A] [answered]
Hi there. I'd like to be able to swithc between buffer focus and nvimtree focus without toggling the tree needlessly. How do i find out the state of the tree? What's the best way to accomplish this otherwise?
In pseudocode, im roughly trying to do something like this :
```
function my_utility()
    if (nvimtree_is_open) 
    
             vim.cmd("NvimTreeFocus")
    else:
              vim.cmd("NvimTreeToggle")
              vim.cmd("NvimTreeFocus")
end
```

Thank you

**Comment by alex-courtis** [comment, marked as answer]
You can use events to determine the state: `:help nvim-tree-events`

You could listen for `Event.TreeOpen` and `Event.TreeClose` and record state. It would be best to maintain state per tab.

**Comment by alex-courtis** [comment]
Perhaps I'm not understanding your use case, however it seems that the decision is not necessary: `NvimTreeFocus` will open the tree if it's closed.

**Reply by rtviii** [reply]
Thanks, Alex.  I should have tried that, this solves my issue. It's really good to know about the events too so im glad i asked.

Thanks for your work!

**Reply by alex-courtis** [reply]
Awesome. Please mark as answer ;)
